### PR TITLE
Return error instead of panic while generating delta.

### PR DESCRIPTION
**Pull request by ashleshaAtrey**
Signed-off-by: Ashlesha Atrey <ashlesha.atrey@intel.com>

### CODE REVIEW

File: swupd/manifest.go [hunk 1]
```diff
@@ -580,8 +580,10 @@ func linkDeltaPeersForPack(c *config, oldManifest, newManifest *Manifest) error
 	}
 
 	// Run rename detection on old and new manifests
-	renameDetection(newManifest, added, removed, *c)
-
+	err := renameDetection(newManifest, added, removed, *c)
+	if err != nil {
+		return err
+	}
```
Comment (rchiossi) [inline swupd/manifest.go]: you can just call `return renameDetection(...)` instead.

File: swupd/rename_test.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func TestRename(t *testing.T) {
 	for _, tc := range tests {
 		add := filelist(t, sn, tc.add)
 		remove := markdelete(filelist(t, sn, tc.remove))
-		renameDetection(&Manifest{}, add, remove, config{})
+		_ = renameDetection(&Manifest{}, add, remove, config{})
```
Comment (rchiossi) [inline swupd/rename_test.go]: You should check the error and fail the test if it is not `nil`

File: swupd/manifest.go (hunk 1 above)
Comment (ashleshaAtrey) [reply swupd/manifest.go]: Updated.

File: swupd/rename_test.go (hunk 2 above)
Comment (ashleshaAtrey) [reply swupd/rename_test.go]: Updated.

**Pull request merged**